sport/test/football/feed/CompetitionsTest.scala

```diff
@@ -0,0 +1,232 @@
+
+import java.time.{Clock, Duration, LocalTime, ZoneId, ZonedDateTime}
+import java.time.format.DateTimeFormatter
+import org.scalatest._
+import feed.Competitions
+import football.model.CompetitionTestData
+import model.Competition
+import pa.{
+  Fixture,
+  FootballMatch,
+  LeagueStats,
+  LeagueTableEntry,
+  LeagueTeam,
+  MatchDay,
+  MatchDayTeam,
+  Result,
+  Round,
+  Stage,
+}
+
+class CompetitionsTest extends FreeSpec with Matchers with OptionValues {
+  "Competitions" - {
+    Seq(
+      (Duration.ofSeconds(1), true),
+      (Duration.ofMinutes(1), true),
+      (Duration.ofMinutes(4).plusSeconds(59), true),
+      (Duration.ofMinutes(5), false),
+      (Duration.ofMinutes(10), false),
+    ) foreach { testcase =>
+      (s"isAMatchInProgress returns ${testcase._2} if there's a match that started ${testcase._1} ago") in {
+        val matches: Seq[FootballMatch] =
+          FootballTestData.matchesWithLiveMatchAtCurrentMinusDuration(testcase._1)
+        val testCompetition = FootballTestData.competitions(1).copy(matches = matches)
+        val competitions = Competitions(Seq(testCompetition))
+
+        val isAMatchInProgress = competitions.isAMatchInProgress(competitions.matches, FootballTestData.clock)
+
+        isAMatchInProgress should equal(testcase._2)
+      }
+    }
+
+    Seq(
+      (Duration.ofSeconds(1), true),
+      (Duration.ofMinutes(1), true),
+      (Duration.ofMinutes(4).plusSeconds(59), true),
+      (Duration.ofMinutes(5), false),
+      (Duration.ofMinutes(10), false),
+    ) foreach { testcase =>
+      (s"isAMatchInProgress returns ${testcase._2} if there's a match that will in ${testcase._1}") in {
+        val matches: Seq[FootballMatch] =
+          FootballTestData.matchesWithLiveMatchAtCurrentPlusDuration(testcase._1)
+        val testCompetition = FootballTestData.competitions(1).copy(matches = matches)
+        val competitions = Competitions(Seq(testCompetition))
+        val test = FootballTestData.today
+        val isAMatchInProgress = competitions.isAMatchInProgress(competitions.matches, FootballTestData.clock)
+
+        isAMatchInProgress should equal(testcase._2)
+      }
+    }
+  }
+
+  // I had trouble using the original FootballTestData because it needed testFootballClient
+  // And in my test I didn't need any client at all. As it's testing the Competitions methods
+  // For now I copied the things from the original FootballTestData until I figure out how to fix this
+  // There has also been some changes compared to the original one such as adding clock,
+  // updating matchDay signature to have isLive as parameter and adding two methods
+  // matchesWithLiveMatchAtCurrentMinusDuration & matchesWithLiveMatchAtCurrentPlusDuration
+  object FootballTestData {
+    private val zone = ZoneId.of("Europe/London")
+    val today = ZonedDateTime.now().withZoneSameInstant(zone)
+
+    val clock = {
+      val fixedDate = today.toInstant
+      Clock.fixed(fixedDate, zone)
+    }
```

Comment:
We added this clock as `isAMatchInProgress` needs it and by `fixedDate` the current time can be controlled in the tests